hibernate-core/src/main/java/org/hibernate/sql/results/graph/entity/AbstractEntityInitializer.java

```diff
@@ -850,8 +850,6 @@ private void initializeEntityInstance(Object toInitialize, RowProcessingState ro
 			);
 		}
 
-		assert concreteDescriptor.getIdentifier( entityInstance, session ) != null;
```

Comment:
Shouldn't this rather be the following?

```suggestion
		assert concreteDescriptor.getIdentifier( toInitialize, session ) != null;
```

Reply:
let me try